content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
+}
+def getFoo() {
+    return this.foo;
+}
+def say(name) {
+    echo "Hello world, ${name}"
+}
+----
+
+Then the Pipeline Script can invoke these methods, rooted off the `acme`
+object:
+
+[source,groovy]
+----
+acme.foo = "5";
+echo acme.foo; // print 5
+acme.say "Joe" // print "Hello world, Joe"
+----
+
+[NOTE]
+====
+A variable defined in a shared library will only show up in _Global Variables
+Reference_ (under _Pipeline Syntax_) after you have first run a successful
+build using that library, allowing its sources to be checked out by Jenkins.
+====
 
-Shared Libraries can also define functions which look and feel like built-in
-steps, such as `sh` or `git`.
+=== Defining steps
 
+Shared Libraries can also define global variables which look and feel like
+built-in steps, such as `sh` or `git`.
 
-For example, to define `helloWorld` step, the file `vars/helloWorld.groovy`
-should be created and have a `call` method defined:
+For example, to define `helloWorld`, the file `vars/helloWorld.groovy`
```

Comment:
IIRC they need to be camelCased rather than PascalCased – confirm, and if so, add a `NOTE`.